src/queueing_honey_badger.rs

```diff
@@ -34,12 +36,57 @@ use transaction_queue::TransactionQueue;
 
 pub use dynamic_honey_badger::{Change, ChangeState, Input};
 
-error_chain!{
-    links {
-        DynamicHoneyBadger(dynamic_honey_badger::Error, dynamic_honey_badger::ErrorKind);
+#[derive(Debug, Fail)]
+pub enum ErrorKind {
+    #[fail(display = "Input error: {}", _0)]
+    Input(dynamic_honey_badger::Error),
+    #[fail(display = "Handle message error: {}", _0)]
+    HandleMessage(dynamic_honey_badger::Error),
+    #[fail(display = "Propose error: {}", _0)]
+    Propose(dynamic_honey_badger::Error),
+}
+
+#[derive(Debug)]
+pub struct Error {
+    inner: Context<ErrorKind>,
+}
+
+impl Fail for Error {
+    fn cause(&self) -> Option<&Fail> {
+        self.inner.cause()
+    }
+
+    fn backtrace(&self) -> Option<&Backtrace> {
+        self.inner.backtrace()
     }
 }
 
+impl Error {
+    pub fn kind(&self) -> &ErrorKind {
+        self.inner.get_context()
+    }
+}
+
+impl From<ErrorKind> for Error {
+    fn from(kind: ErrorKind) -> Error {
+        Error { inner: Context::new(kind) }
+    }
+}
+
+impl From<Context<ErrorKind>> for Error {
+    fn from(inner: Context<ErrorKind>) -> Error {
+        Error { inner: inner }
```

Comment:
See the clippy lint.

Reply:
It's probably going to say don't `inner: inner`. I don't use clippy :P